conf/webknossos.latest.routes

```diff
@@ -141,6 +141,13 @@ GET           /annotations/:typ/:id/sharedTeams
 PATCH         /annotations/:typ/:id/sharedTeams                                         controllers.AnnotationController.updateSharedTeams(typ: String, id: String)
 PATCH         /annotations/:typ/:id/othersMayEdit                                       controllers.AnnotationController.updateOthersMayEdit(typ: String, id: String, othersMayEdit: Boolean)
 
+# Annotation Private Links
+GET           /annotations/zarr_private/byAccessToken/:accessId                         controllers.AnnotationPrivateLinkController.lookupPrivateLink(accessId: String)
+POST          /annotations/zarr_private                                                 controllers.AnnotationPrivateLinkController.create
+PUT           /annotations/zarr_private/:id                                             controllers.AnnotationPrivateLinkController.update(id: String)
+DELETE        /annotations/zarr_private/:id                                             controllers.AnnotationPrivateLinkController.delete(id: String)
+GET           /annotations/zarr_private/:id                                             controllers.AnnotationPrivateLinkController.get(id: String)
```

Comment:
There should also be a list route. It should list all that the user is allowed see (note my other comment about the readAccessQuery)

Reply:
good point, we should go for camel case. Do you have an idea for a good name? just zarrPrivate? Or zarrPrivateLinks?

Reply:
how about just „zarr“?

Reply:
I think I’d rather go for `zarrPrivateLinks`, `privateLinks`, or `zarrAccessToken` or something along those lines here, since that is the object that the CRUD routes are dealing with. Note that this is not the user-facing URI that they paste into the external tool; this is only used by wk.

Those user-facing URIs should probably just say `zarr` or possibly `annotations/zarr/` or 'zarr/annotations/`, if it does not lead to conflicts with the existing dataset streaming routes

Reply:
I thought you were talking about user facing URIs.

![](https://media.giphy.com/media/COYGe9rZvfiaQ/giphy.gif)

Reply:
There is also no other route that uses a second "name" after the initial "name" (usually it is only `annotations`). I think it should just be `/zarrPrivateLinks` instead of `annotations/zarrPrivateLinks`.